ddtrace/tracer/rules_sampler.go

```diff
@@ -589,78 +590,67 @@ func samplingRulesFromEnv() (trace, span []SamplingRule, err error) {
 	return trace, span, err
 }
 
-// unmarshalSamplingRules unmarshals JSON from b and returns the sampling rules found, attributing
-// the type t to them. If any errors are occurred, they are returned.
-func unmarshalSamplingRules(b []byte, spanType SamplingRuleType) ([]SamplingRule, error) {
+func (sr *SamplingRule) UnmarshalJSON(b []byte) error {
 	if len(b) == 0 {
-		return nil, nil
+		return nil
 	}
-	var jsonRules []struct {
+	var jsonRule struct {
 		Service      string            `json:"service"`
 		Name         string            `json:"name"`
 		Rate         json.Number       `json:"sample_rate"`
 		MaxPerSecond float64           `json:"max_per_second"`
 		Resource     string            `json:"resource"`
 		Tags         map[string]string `json:"tags"`
 	}
-	err := json.Unmarshal(b, &jsonRules)
+	err := json.Unmarshal(b, &jsonRule)
 	if err != nil {
+		return fmt.Errorf("error unmarshalling JSON: %v", err)
+	}
+	if jsonRule.Rate == "" {
+		jsonRule.Rate = "1"
+	}
+	rate, err := jsonRule.Rate.Float64()
+	if err != nil {
+		return err
+	}
+	if rate < 0.0 || rate > 1.0 {
+		return fmt.Errorf("rate is out of [0.0, 1.0] range")
+	}
+	tagGlobs := make(map[string]*regexp.Regexp, len(jsonRule.Tags))
+	for k, g := range jsonRule.Tags {
+		tagGlobs[k] = globMatch(g)
+	}
+	sr.Service = globMatch(jsonRule.Service)
+	sr.Name = globMatch(jsonRule.Name)
+	sr.Rate = rate
+	sr.MaxPerSecond = jsonRule.MaxPerSecond
+	sr.Resource = globMatch(jsonRule.Resource)
+	sr.Tags = tagGlobs
+
+	return nil
+}
+
+// unmarshalSamplingRules unmarshals JSON from b and returns the sampling rules found, attributing
+// the type t to them. If any errors are occurred, they are returned.
+func unmarshalSamplingRules(b []byte, spanType SamplingRuleType) ([]SamplingRule, error) {
+	if len(b) == 0 {
+		return nil, nil
+	}
```

Comment:
This function is being used when unmarshaling code from the env variables, so you shouldn't remove all the validation from here.